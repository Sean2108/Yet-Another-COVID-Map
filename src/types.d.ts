export interface CaseCounts {
  [country: string]: {
    [state: string]: {
      Lat: number;
      Long: number;
      Confirmed: number;
      Deaths: number;
    };
  };
}

export interface GeoJson {
  type: string;
  features: Array<GeoJsonFeature>;
}

export interface GeoJsonFeature {
  type: string;
  properties: {
    country: string;
    state: string;
    value: number;
  };
  geometry: {
    type: string;
    coordinates: Array<number>;
  };
}

export interface CaseCountCapitalised {
  Date: string;
  Confirmed: number;
  Deaths: number;
}

export interface CaseCount {
  date: Date | null;
  confirmed: number;
  deaths: number;
}

export interface NewsItem {
    Source: string;
    Title: string;
    Description: string;
    URL: string;
    ThumbnailURL: string;
    PublishedAt: string;
}
