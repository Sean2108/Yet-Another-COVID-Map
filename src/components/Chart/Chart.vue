<template>
  <div :id="id" />
</template>

<script src="./Chart.ts" />
