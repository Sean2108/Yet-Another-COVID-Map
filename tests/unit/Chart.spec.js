import { shallowMount, createLocalVue } from "@vue/test-utils";
import Chart from "../../src/components/Chart.vue";
import * as d3 from "d3";
import Vuetify from "vuetify";
import * as utils from "../../src/utils";

utils.fetchData = jest.fn();

describe("Chart.vue", () => {
  let localVue, wrapper;
  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuetify);

    wrapper = shallowMount(Chart, {
      localVue,
      attachToDocument: true,
      propsData: {
        id: "testChart",
        chartData: [
          { date: "2/1/20", confirmed: 20, deaths: 5, recovered: 10 },
          { date: "2/2/20", confirmed: 50, deaths: 15, recovered: 25 },
          { date: "2/3/20", confirmed: 75, deaths: 15, recovered: 40 }
        ],
        width: 150,
        height: 100,
        legendX: 120,
        legendY: 50,
        marginLeft: 30,
        marginRight: 50,
        showPercentages: false
      }
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("should draw graph correctly", () => {
    expect(d3.select("svg")).not.toBeNull();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should update graph on data update", async () => {
    wrapper.setProps({
      chartData: [
        { date: "2/1/20", confirmed: 25, deaths: 10, recovered: 15 },
        { date: "2/2/20", confirmed: 55, deaths: 10, recovered: 35 },
        { date: "2/3/20", confirmed: 80, deaths: 20, recovered: 60 }
      ]
    });
    await localVue.nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should update graph on showPercentages update", async () => {
    wrapper.setProps({
      showPercentages: true
    });
    await localVue.nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should alter svg correctly on mouse actions", async () => {
    wrapper.vm.onMouseEnter(d3.select("svg"), "red");
    await localVue.nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot();
    wrapper.vm.onMouseLeave(d3.select("svg"), "red");
    await localVue.nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
