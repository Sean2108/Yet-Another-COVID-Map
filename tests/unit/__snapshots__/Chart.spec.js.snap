// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart.vue should alter svg correctly on mouse actions 1`] = `
<div
  id="testChart"
>
  <svg
    height="100"
    width="150"
  >
    <g
      transform="translate(30,10)"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0,70)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H70.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            February
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(18,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(35.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Feb 02
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(53,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(70.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Mon 03
          </text>
        </g>
      </g>
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
      >
        <path
          class="domain"
          d="M-6,70.5H0.5V0.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,70.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,61.16666666666667)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,51.833333333333336)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,42.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,33.166666666666664)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            40
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,23.833333333333336)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            50
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,14.499999999999996)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,5.166666666666666)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            70
          </text>
        </g>
      </g>
      <path
        class="chartlines"
        d="M0,51.333333333333336C11.666666666666666,41.611111111111114,23.333333333333336,31.888888888888893,35,23.333333333333336C46.666666666666664,14.77777777777778,58.333333333333336,7.388888888888888,70,0"
        fill="none"
        id="line-cyan"
        stroke="cyan"
        stroke-width="1.5"
        style="stroke-width: 0.5;"
      />
      <path
        class="chartlines"
        d="M0,65.33333333333333C11.666666666666666,60.666666666666664,23.333333333333336,56,35,56C46.666666666666664,56,58.333333333333336,56,70,56"
        fill="none"
        id="line-red"
        stroke="red"
        stroke-width="1.5"
        style="stroke-width: 3;"
      />
      <path
        class="chartlines"
        d="M0,60.66666666666667C11.666666666666666,56.00000000000001,23.333333333333336,51.333333333333336,35,46.66666666666667C46.666666666666664,42.00000000000001,58.333333333333336,37.333333333333336,70,32.666666666666664"
        fill="none"
        id="line-lightgreen"
        stroke="lightgreen"
        stroke-width="1.5"
        style="stroke-width: 0.5;"
      />
      <path
        class="chartlines"
        d="M0,65.33333333333333C11.666666666666666,64.16666666666667,23.333333333333336,63,35,60.66666666666667C46.666666666666664,58.33333333333334,58.333333333333336,54.833333333333336,70,51.333333333333336"
        fill="none"
        id="line-orange"
        stroke="orange"
        stroke-width="1.5"
        style="stroke-width: 0.5;"
      />
      <circle
        class="legendcircles"
        cx="120"
        cy="50"
        id="circle-cyan"
        r="6"
        style="fill: cyan;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="50"
      >
        Confirmed cases
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="70"
        id="circle-red"
        r="8"
        style="fill: red; stroke: black;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="70"
      >
        Deaths
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="90"
        id="circle-lightgreen"
        r="6"
        style="fill: lightgreen;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="90"
      >
        Recoveries
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="110"
        id="circle-orange"
        r="6"
        style="fill: orange;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="110"
      >
        Active cases
      </text>
    </g>
  </svg>
</div>
`;

exports[`Chart.vue should alter svg correctly on mouse actions 2`] = `
<div
  id="testChart"
>
  <svg
    height="100"
    width="150"
  >
    <g
      transform="translate(30,10)"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0,70)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H70.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            February
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(18,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(35.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Feb 02
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(53,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(70.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Mon 03
          </text>
        </g>
      </g>
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
      >
        <path
          class="domain"
          d="M-6,70.5H0.5V0.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,70.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,61.16666666666667)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,51.833333333333336)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,42.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,33.166666666666664)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            40
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,23.833333333333336)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            50
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,14.499999999999996)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,5.166666666666666)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            70
          </text>
        </g>
      </g>
      <path
        class="chartlines"
        d="M0,51.333333333333336C11.666666666666666,41.611111111111114,23.333333333333336,31.888888888888893,35,23.333333333333336C46.666666666666664,14.77777777777778,58.333333333333336,7.388888888888888,70,0"
        fill="none"
        id="line-cyan"
        stroke="cyan"
        stroke-width="1.5"
        style="stroke-width: 1.5;"
      />
      <path
        class="chartlines"
        d="M0,65.33333333333333C11.666666666666666,60.666666666666664,23.333333333333336,56,35,56C46.666666666666664,56,58.333333333333336,56,70,56"
        fill="none"
        id="line-red"
        stroke="red"
        stroke-width="1.5"
        style="stroke-width: 1.5;"
      />
      <path
        class="chartlines"
        d="M0,60.66666666666667C11.666666666666666,56.00000000000001,23.333333333333336,51.333333333333336,35,46.66666666666667C46.666666666666664,42.00000000000001,58.333333333333336,37.333333333333336,70,32.666666666666664"
        fill="none"
        id="line-lightgreen"
        stroke="lightgreen"
        stroke-width="1.5"
        style="stroke-width: 1.5;"
      />
      <path
        class="chartlines"
        d="M0,65.33333333333333C11.666666666666666,64.16666666666667,23.333333333333336,63,35,60.66666666666667C46.666666666666664,58.33333333333334,58.333333333333336,54.833333333333336,70,51.333333333333336"
        fill="none"
        id="line-orange"
        stroke="orange"
        stroke-width="1.5"
        style="stroke-width: 1.5;"
      />
      <circle
        class="legendcircles"
        cx="120"
        cy="50"
        id="circle-cyan"
        r="6"
        style="fill: cyan; stroke: transparent;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="50"
      >
        Confirmed cases
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="70"
        id="circle-red"
        r="6"
        style="fill: red; stroke: transparent;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="70"
      >
        Deaths
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="90"
        id="circle-lightgreen"
        r="6"
        style="fill: lightgreen; stroke: transparent;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="90"
      >
        Recoveries
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="110"
        id="circle-orange"
        r="6"
        style="fill: orange; stroke: transparent;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="110"
      >
        Active cases
      </text>
    </g>
  </svg>
</div>
`;

exports[`Chart.vue should draw graph correctly 1`] = `
<div
  id="testChart"
>
  <svg
    height="100"
    width="150"
  >
    <g
      transform="translate(30,10)"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0,70)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H70.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            February
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(18,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(35.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Feb 02
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(53,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(70.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Mon 03
          </text>
        </g>
      </g>
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
      >
        <path
          class="domain"
          d="M-6,70.5H0.5V0.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,70.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,61.16666666666667)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,51.833333333333336)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,42.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,33.166666666666664)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            40
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,23.833333333333336)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            50
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,14.499999999999996)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,5.166666666666666)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            70
          </text>
        </g>
      </g>
      <path
        class="chartlines"
        d="M0,51.333333333333336C11.666666666666666,41.611111111111114,23.333333333333336,31.888888888888893,35,23.333333333333336C46.666666666666664,14.77777777777778,58.333333333333336,7.388888888888888,70,0"
        fill="none"
        id="line-cyan"
        stroke="cyan"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,65.33333333333333C11.666666666666666,60.666666666666664,23.333333333333336,56,35,56C46.666666666666664,56,58.333333333333336,56,70,56"
        fill="none"
        id="line-red"
        stroke="red"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,60.66666666666667C11.666666666666666,56.00000000000001,23.333333333333336,51.333333333333336,35,46.66666666666667C46.666666666666664,42.00000000000001,58.333333333333336,37.333333333333336,70,32.666666666666664"
        fill="none"
        id="line-lightgreen"
        stroke="lightgreen"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,65.33333333333333C11.666666666666666,64.16666666666667,23.333333333333336,63,35,60.66666666666667C46.666666666666664,58.33333333333334,58.333333333333336,54.833333333333336,70,51.333333333333336"
        fill="none"
        id="line-orange"
        stroke="orange"
        stroke-width="1.5"
      />
      <circle
        class="legendcircles"
        cx="120"
        cy="50"
        id="circle-cyan"
        r="6"
        style="fill: cyan;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="50"
      >
        Confirmed cases
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="70"
        id="circle-red"
        r="6"
        style="fill: red;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="70"
      >
        Deaths
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="90"
        id="circle-lightgreen"
        r="6"
        style="fill: lightgreen;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="90"
      >
        Recoveries
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="110"
        id="circle-orange"
        r="6"
        style="fill: orange;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="110"
      >
        Active cases
      </text>
    </g>
  </svg>
</div>
`;

exports[`Chart.vue should update graph on data update 1`] = `
<div
  id="testChart"
>
  <svg
    height="100"
    width="150"
  >
    <g
      transform="translate(30,10)"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0,70)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H70.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            February
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(18,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(35.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Feb 02
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(53,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(70.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Mon 03
          </text>
        </g>
      </g>
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
      >
        <path
          class="domain"
          d="M-6,70.5H0.5V0.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,70.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,61.75)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,53)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,44.25)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,35.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            40
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,26.75)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            50
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,18)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,9.25)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            70
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            80
          </text>
        </g>
      </g>
      <path
        class="chartlines"
        d="M0,48.125C11.666666666666666,39.01041666666667,23.333333333333336,29.895833333333332,35,21.875C46.666666666666664,13.854166666666668,58.333333333333336,6.927083333333333,70,0"
        fill="none"
        id="line-cyan"
        stroke="cyan"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,61.25C11.666666666666666,61.25,23.333333333333336,61.25,35,61.25C46.666666666666664,61.25,58.333333333333336,56.875,70,52.5"
        fill="none"
        id="line-red"
        stroke="red"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,56.875C11.666666666666666,51.40625,23.333333333333336,45.9375,35,39.375C46.666666666666664,32.8125,58.333333333333336,25.15625,70,17.5"
        fill="none"
        id="line-lightgreen"
        stroke="lightgreen"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,70C11.666666666666666,65.625,23.333333333333336,61.25,35,61.25C46.666666666666664,61.25,58.333333333333336,65.625,70,70"
        fill="none"
        id="line-orange"
        stroke="orange"
        stroke-width="1.5"
      />
      <circle
        class="legendcircles"
        cx="120"
        cy="50"
        id="circle-cyan"
        r="6"
        style="fill: cyan;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="50"
      >
        Confirmed cases
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="70"
        id="circle-red"
        r="6"
        style="fill: red;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="70"
      >
        Deaths
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="90"
        id="circle-lightgreen"
        r="6"
        style="fill: lightgreen;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="90"
      >
        Recoveries
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="110"
        id="circle-orange"
        r="6"
        style="fill: orange;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="110"
      >
        Active cases
      </text>
    </g>
  </svg>
</div>
`;

exports[`Chart.vue should update graph on showPercentages update 1`] = `
<div
  id="testChart"
>
  <svg
    height="100"
    width="150"
  >
    <g
      transform="translate(30,10)"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0,70)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H70.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            February
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(18,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(35.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Feb 02
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(53,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            12 PM
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(70.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            Mon 03
          </text>
        </g>
      </g>
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
      >
        <path
          class="domain"
          d="M-6,70.5H0.5V0.5H-6"
          stroke="currentColor"
        />
      </g>
      <path
        class="chartlines"
        d="M0,NaNC11.666666666666666,NaN,23.333333333333336,NaN,35,NaNC46.666666666666664,NaN,58.333333333333336,NaN,70,NaN"
        fill="none"
        id="line-red"
        stroke="red"
        stroke-width="1.5"
      />
      <path
        class="chartlines"
        d="M0,NaNC11.666666666666666,NaN,23.333333333333336,NaN,35,NaNC46.666666666666664,NaN,58.333333333333336,NaN,70,NaN"
        fill="none"
        id="line-lightgreen"
        stroke="lightgreen"
        stroke-width="1.5"
      />
      <circle
        class="legendcircles"
        cx="120"
        cy="50"
        id="circle-red"
        r="6"
        style="fill: red;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="50"
      >
        Fatality rate
      </text>
      <circle
        class="legendcircles"
        cx="120"
        cy="70"
        id="circle-lightgreen"
        r="6"
        style="fill: lightgreen;"
      />
      <text
        alignment-baseline="middle"
        style="font-size: 10px; fill: #fff;"
        x="130"
        y="70"
      >
        Recovery rate
      </text>
    </g>
  </svg>
</div>
`;
